h2o-py/tests/testdir_algos/gam/pyunit_PUBDEV_7367_random_gridsearch_constraints.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import division
+from __future__ import print_function
+from past.utils import old_div
+import sys
+sys.path.insert(1, "../../../")
+import h2o
+from tests import pyunit_utils
+from h2o.estimators.gam import H2OGeneralizedAdditiveEstimator
+from h2o.grid.grid_search import H2OGridSearch
+
+# In this test, we check to make sure that a random grid search on a GAM works with constrained hyperparameters.
+# The test compares the results of the grid search models with the models we created
+# by manually searching over the hyperspace.
+# If the coefficients do not match or an incorrect number of models is generated, the test throws an assertion error.
+class test_random_gam_gridsearch_specific:
+
+    h2o_data = []
+    myX = []
+    myY = []
+    h2o_model = []
+    search_criteria = {'strategy': 'RandomDiscrete', "max_models": 24, "seed": 1}
+    hyper_parameters = {'lambda': [0, 0.01],
+                        'constraints': [{'scale': [[1, 1, 1], [2, 2, 2]],
+                                         'num_knots': [[5, 5, 5], [5, 6, 7]],
+                                         'gam_columns': [["C6", "C7", "C8"]]},
+                                        {'scale': [[1, 1], [2, 2]],
+                                         'num_knots': [[5, 5], [6, 6]],
+                                         'gam_columns': [["C6", "C7"], ["C7", "C8"]]}]}
+    manual_gam_models = []
+    num_grid_models = 0
```

Comment:
Karhtik:  Please add one more constraints:
{'lambda_search':False, 'alpha':[0.9,0.1]}

Remember to fix the maximum number of models expected.